pynxtools/dataconverter/readers/transmission/reader.py

```diff
@@ -149,25 +146,36 @@ def read_detectors(metadata: list) -> Dict[str, Any]:
     detector_times = parse_detector_line(metadata[32], float)
     detector_gains = parse_detector_line(metadata[35], float)
     detector_changes = [float(x) for x in metadata[43].split()]
-    wavelength_ranges = \
+    wavelength_ranges = (
         [mpars.MIN_WAVELENGTH] + detector_changes[::-1] + [mpars.MAX_WAVELENGTH]
+    )
 
     template.update(
         convert_detector_to_template(
-            "PMT", detector_slits[2], detector_times[2],
-            None, 2, [wavelength_ranges[0], wavelength_ranges[1]]
+            "PMT",
+            detector_slits[-1],
```

Comment:
This is also relevant to use -1 instead of 2 here. This line was actually failing if there was only one detector